CONTRIBUTING.md

```diff
@@ -0,0 +1,70 @@
+# Contributing
+
+- [Contributing](#contributing)
+  - [Architecture Decision Records (ADR)](#architecture-decision-records-adr)
+  - [PR Targeting](#pr-targeting)
+  - [Development Procedure](#development-procedure)
+  - [Semantic Versioning](#semantic-versioning)
+  - [Testing](#testing)
+  - [Protobuf](#protobuf)
+
+Thank you for considering making contributions to ICS!
+
+To contribute, you could start by browsing [new issues](https://github.com/cosmos/interchain-security/issues).
+
+## Architecture Decision Records (ADR)
+
+When proposing an architecture decision for ICS, please start by opening an [issue](https://github.com/cosmos/interchain-security/issues/new/choose) with a summary of the proposal. Once the proposal has been discussed and there is rough alignment on a high-level approach to the design, you may either start development, or write an ADR.
+
+If your architecture decision is a simple change, you may contribute directly without writing an ADR. However, if you are proposing a significant change, please include a corresponding ADR.
+
+In certain circumstances, the architecture decision may require changes to the ICS spec. Note that the spec is responsible for defining language-agnostic, implementation-agnostic behaviors for the ICS protocol. Whereas ADRs are responsible for communicating implementation decisions contained within this repo.
+
+To create an ADR, follow the [template](https://github.com/cosmos/interchain-security/blob/main/docs/architecture/adr-template.md) and [doc](https://github.com/cosmos/interchain-security/blob/main/docs/architecture/README.md). If you would like to see examples of how these are written, please refer to the current [ADRs](https://github.com/cosmos/interchain-security/tree/main/docs/architecture).
+
+### ADR Proposals
+
+Architecture Decision Records (ADRs) may be proposed by any contributors or maintainers of ICS. ADRs are intended to be iterative, and may be merged into `main` while still in a `Proposed` status
+
+## PR Targeting
+
+ICS adheres to the [trunk based development branching model](https://trunkbaseddevelopment.com/).
+
+All feature additions and all bug fixes must be targeted against `main`. Exception is for bug fixes which may only be relevant to a previously released version. In that case, the bug fix PR must target against the release branch.
+
+If needed, we will backport a commit from `main` to a release branch with appropriate consideration of versioning.
```

Comment:
Dissimilar to gaia, I'm proposing that we do not create running release branches until they're actually needed. Ie. if version X.Y.Z is the latest development version, and we want all recent changes applied to that version, it doesn't need a release branch yet. 

We need the release branch only when the release is cut and integrated into a gaia proposal. Additional commits on running release branches should be rare, and would usually fix a critical bug

Reply:
It's also notable that a bugfix to an ICS release branch could often constitute a MAJOR version bump. Meaning it wouldn't really make sense to maintain a `release/v1.x.x` branch (in that a bug fix would make it version 2.x.x) 

We could maintain release branches based on the date that release first launched. `release/03-15-23` for example. Semver would still exist on these branches, just as tags. 

The only edge case left to decide on, is if a critical bug shows up in prod, 0-2 weeks before an upgrade proposal passes, and we have to fix an old release version, we might have a naming conflict the version being voted on.

Lmk what you think @jtremback @mpoke